convert2rhel/utils.py

```diff
@@ -620,4 +627,59 @@ def set_locale():
     os.environ.update({"LC_ALL": "C", "LANG": "C"})
 
 
+def options_from_config_files(cfg_path=None):
+    """Get the options from config file.
+
+    The order is:
+    1. if filename is set then load from it
+    2. ~/.convert2rhel.ini
+    3. /etc/convert2rhel.ini
+
+    Return dict of supported options with their values.
+    """
+    headers = ["subscription_manager"]  # supported sections in config file
+    config_file = configparser.ConfigParser()
+    paths = CONFIG_PATHS
+    # Create dict with all supproted options, all of them set to None
+    # needed for avoiding problems with files priority
+    # The name of supported option MUST correspond with the name in ToolOpts()
+    # Otherwise it won't be used
+    supported_opts = {"password": None, "activation_key": None}
+
+    if cfg_path:
+        if not os.path.exists(cfg_path):
+            raise OSError(2, "No such file or directory: '" + cfg_path + "'")
+        paths.insert(0, cfg_path)  # highest priority
+
+    for path in paths:
```

Comment:
Something I just noticed: `paths` needs to have `os.path.expanduser()` run on the paths contained in it.  Otherwise when we call `os.path.exist()` later, it will think that it's looking for a literal `~/.convert2rhel.ini`, not for `/home/USERNAME/.convert2rhel.ini`.

There's two places we could call `expanduser()`.. Here, or where we define `CONFIG_PATHS`. I lean towards doing it here for the following reason: If cfg_path contains a path with an initial tilde, then it should probably also be expanded. My justification isn't really strong because, in most cases, the user will be specifying cfg_path via a command line argument and the user's shell will do that expansion but it could occur if `options_from_cfg_files()` is called with a cfg_path that isn't from the command line (for instance, a unit test) or someone needs to run convert2rhel programmatically (like `subprocess.Popen(["convert2rhel", "--cfg_file", "~badger/my-convert2rhel.ini"])` ) .

My suggestion for doing it here would be:
```suggestion
    for path in (os.path.join(p) for p in paths):
```